derive/src/to_delegate.rs

```diff
@@ -169,9 +171,7 @@ impl RpcMethod {
 			special_args.iter().find(|(_,sty)| sty == ty).is_none());
 
 		let tuple_fields : &Vec<_> =
-			&(0..param_types.len() as u8)
-				.map(|x| ident(&((x + 'a' as u8) as char).to_string()))
-				.collect();
+			&(TUPLE_FIELD_NAMES.iter().take(param_types.len()).map(|name| ident(name)).collect());
```

Comment:
@kirushik I've found it - max tuple size for which `Deserialize` is implemented in serde is 16: https://docs.serde.rs/serde/trait.Deserialize.html#impl-Deserialize%3C%27de%3E.

So we should check the length of `param_types` (< 16) and return an error. Though we can do that in a separate PR if you like since this one is about clippy warnings - and at least your changes here will make it fail in a less weird way.

Reply:
I've implemented the check, see https://github.com/paritytech/jsonrpc/pull/392#discussion_r259466976

Reply:
@ordian my apologies I completely missed that